src/transformers/pipelines/text_generation.py

```diff
@@ -391,8 +413,19 @@ def postprocess(self, model_outputs, return_type=ReturnType.FULL_TEXT, clean_up_
                         all_text = prompt_text + all_text
                     elif isinstance(prompt_text, Chat):
                         # Explicit list parsing is necessary for parsing chat datasets
-                        all_text = list(prompt_text.messages) + [{"role": "assistant", "content": all_text}]
-
+                        if continue_final_message is None:
+                            continue_final_message = prompt_text.messages[-1]["role"] == "assistant"
```

Comment:
Maybe add a comment explaining what is being done under the hood here (so assuming that the request is to continue the final message if the last message was already sent by the assistant?)

Reply:
Done!